phoenix-core/src/main/java/org/apache/phoenix/replication/SystemCatalogWALEntryFilter.java

```diff
@@ -103,18 +104,20 @@ private boolean isTenantRowCellSystemChildLink(final Cell cell) {
     if (!isTenantRowCell) {
       boolean isChildLink = CellUtil.matchingQualifier(
         cell, PhoenixDatabaseMetaData.LINK_TYPE_BYTES);
-      if (isChildLink) {
-        if (CellUtil.matchingValue(cell, CHILD_TABLE_BYTES)) {
+      // Check if cell is of type LINK_TYPE with value 4 or DeleteFamily
+      if ((isChildLink && CellUtil.matchingValue(cell, CHILD_TABLE_BYTES)) ||
+          CellUtil.isDeleteFamily(cell) ) {
```

Comment:
this logic means you will replicate whenever it's a DeleteFamily cell, regardless of childlink or not. Is this the intention?

Reply:
To replicate a delete marker we are using two checks.
1. The cell is of type `DeleteFamily`.
2. The length for `COLUMN_NAME` is greater than `0`. (For tenant views column_name is populated with tenant_id which is not the case for global views.)

One may argue that the above two checks are also true for delete markers of column rows and we may over-replicate that as well. But, we are avoiding all those scenarios by doing the above filtering only for the mutations belonging to `SYSTEM.CHILD_LINK` table. With 4.15+ we are assured that `SYSTEM.CHILD_LINK` table only stores parent-child linking rows with `LINK_TYPE=4`. Let me know if you still feel we are missing any scenario. Thanks